tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_horizontal_execution_merging.py

```diff
@@ -160,14 +160,20 @@ def test_on_the_fly_merging_repeated():
         vertical_loops__0__sections__0__horizontal_executions=[
             HorizontalExecutionFactory(body=[AssignStmtFactory(left__name="tmp")]),
             HorizontalExecutionFactory(
-                body=[AssignStmtFactory(left__name="out1", right__name="tmp")]
+                body=[
+                    AssignStmtFactory(
+                        left__name="tmp_offset", right__name="tmp", right__offset__i=1
+                    )
+                ]
+            ),
+            HorizontalExecutionFactory(
+                body=[AssignStmtFactory(left__name="out", right__name="tmp_offset")]
             ),
-            HorizontalExecutionFactory(body=[AssignStmtFactory(left__name="tmp")]),
             HorizontalExecutionFactory(
-                body=[AssignStmtFactory(left__name="out2", right__name="tmp")]
+                body=[AssignStmtFactory(left__name="out2", right__name="tmp_offset")]
             ),
         ],
-        declarations=[TemporaryFactory(name="tmp")],
+        declarations=[TemporaryFactory(name="tmp"), TemporaryFactory(name="tmp_offset")],
```

Comment:
I propose to make this a new separate test, the old one relied on the particular read-write pattern that was there and I'm not sure if the new test would capture the old test properly.

Reply:
Sounds good. The problem was that this test didn't even claim to do what it said it was doing. It does "multiple" applications of the pass, but there is not a nested `_merge` call. I added this test back and renamed both.